### DISCUSSION TITLE
case insenstive map

### DISCUSSION THREAD

**Discussion by austriancoder** [nlohmann/json#3756] [Q&A]
Hi all

I am trying to integrate an case intensive map to store the parsed json. I only need to have the key case insensitive. Here is what I tried.

`
struct lower_compare {
    static bool noase_compare(const unsigned char &a, const unsigned char &b) {
        return std::tolower(a) == std::tolower(b);
    }

    bool operator()(const std::string &lhs, const std::string &rhs) const {
        return std::lexicographical_compare(lhs.begin(), lhs.end(), rhs.begin(), rhs.end(), noase_compare);
    }
};

template <class Key, class T, class IgnoredLess = std::less<Key>,
          class Allocator = std::allocator<std::map<const Key, T>>>
struct case_insenitive_map : std::map<const Key, T, lower_compare> {};`

I am using it that way:

`
using json = nlohmann::basic_json<case_insenitive_map>;
auto j = json::parse("valid json .. ").

`

If I am using e.g. j.contains("kEy") it tells me that the element with that key does not exists.

**Comment by falbrechtskirchinger** [comment]
FIFY:

```cpp
return std::tolower(a) < std::tolower(b);
```
